system/lib/wasmfs/memory_backend.h

```diff
@@ -44,18 +44,19 @@ class MemoryFile : public DataFile {
 class MemoryDirectory : public Directory {
   // Use a vector instead of a map to save code size.
   struct ChildEntry {
-    std::string name;
+    path_string name;
```

Comment:
Would it be possible to keep using `std::string` here and keep the use of `path_string` to the target-independent code only?

Reply:
If `std::string` is used in `ChildEntry` some operations such as `unlink()` won't work correctly.
For example, this line in test will not delete anything:
https://github.com/emscripten-core/emscripten/blob/1d568bcbfcc4818d554d3269f0a742078c2c96a6/test/test_path_icase.c#L93
And the subsequent check that the `subdir` is empty will fail:
` Aborted(Assertion failed: get_dir_files("subdir") == 0, at: emscripten/test/test_path_icase.c,99,main)`

Reply:
Would it be acceptable behavior for WasmFS to normalize all filenames to be all lowercase (or all uppercase) when in case-insensitive mode? That way we could have the backend-independent code perform normalization and backends wouldn't have to think about it.

Actually, if we can do that, then maybe we could even implement case-insensitivity via a virtual backend that performs case normalization before forwarding operations to another backend. That way neither the core code nor the other backends would have to think about case insensitivity! (Except for the node backend, as discussed in that other comment)

Reply:
> Would it be acceptable behavior for WasmFS to normalize all filenames to be all lowercase (or all uppercase) when in case-insensitive mode?

I think it's fine and could save some work. 
But wouldn't that make such calls as `FS.readdir('.')` return only lower-case files?

Reply:
Yes, `FS.readdir` would only return lowercase files if we did that and your example would return false. Maybe this would cause too many problems to be worth it.